File: MyMovies-MVP/app/src/main/java/com/jbrunton/mymovies/api/moviedetails/MovieDetailsActivity.java
package com.jbrunton.mymovies.api.moviedetails;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.widget.TextView;

import com.jbrunton.mymovies.R;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MovieDetailsActivity extends AppCompatActivity {

    private MovieDetailsViewModel viewModel;
    @BindView(R.id.overview) TextView overview;
    @BindView(R.id.toolbar) Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_movie_details);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        setTitle("");

        ButterKnife.bind(this);

        MovieDetailsViewModel.Factory factory = new MovieDetailsViewModel.Factory(
                getIntent().getExtras().getString("MOVIE_ID"));
        viewModel = ViewModelProviders.of(this, factory)
                .get(MovieDetailsViewModel.class);
        viewModel.viewState().observe(this, this::updateView);
    }

    private void updateView(MovieDetailsViewState viewState) {
        if (viewState.movie().isPresent()) {
            setTitle(viewState.movie().get().title());
            overview.setText(viewState.movie().get().overview().get());
        }
        //moviesList.setVisibility(toVisibility(!viewState.showError()));
        //moviesAdapter.setDataSource(viewState.movies());
    }
}

File: app/src/main/java/com/jbrunton/mymovies/ui/moviedetails/MovieDetailsActivity.kt
package com.jbrunton.mymovies.ui.moviedetails

import android.view.View
import androidx.appcompat.widget.Toolbar
import com.jbrunton.mymovies.R
import com.jbrunton.mymovies.libs.ui.controllers.rootView
import com.jbrunton.mymovies.libs.ui.livedata.observe
import com.jbrunton.mymovies.libs.ui.views.BaseActivity
import com.jbrunton.mymovies.libs.ui.viewmodels.injectViewModel

class MovieDetailsActivity : BaseActivity<MovieDetailsViewModel>() {
    val toolbar: Toolbar get() = findViewById(R.id.toolbar)
    val favorite: View get() = findViewById(R.id.favorite)
    val unfavorite: View get() = findViewById(R.id.unfavorite)

    override val viewModel: MovieDetailsViewModel by injectViewModel { movieId() }
    private val viewController = MovieDetailsViewController()

    override fun onCreateLayout() {
        setContentView(R.layout.activity_movie_details)

        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        title = ""

        viewController.onViewCreated(rootView)
    }

    override fun onBindListeners() {
        viewController.error_try_again.setOnClickListener { viewModel.onRetryClicked() }
        favorite.setOnClickListener { viewModel.onFavoriteClicked() }
        unfavorite.setOnClickListener { viewModel.onUnfavoriteClicked() }
    }

    override fun onObserveData() {
        super.onObserveData()
        viewModel.viewState.observe(this, viewController::updateView)
    }

    private fun movieId(): String = intent.extras["MOVIE_ID"] as String
}
